addons/mass_mailing_marketing_card/models/card_card.py

```diff
@@ -0,0 +1,42 @@
+from collections import defaultdict
+
+from odoo import models, osv
+
+
+class CardCard(models.Model):
+    _inherit = 'card.card'
+
+    def _create_if_not_exists(self, card_values_all):
```

Comment:
Don't really get why this is a separate method, seems highly specific to the mass mailing processing, not sure how anyone will re-use this.